test/Interop/Cxx/templates/Inputs/canonical-types.h

```diff
@@ -0,0 +1,19 @@
+#ifndef TEST_INTEROP_CXX_TEMPLATES_INPUTS_CANONICAL_TYPES_H
+#define TEST_INTEROP_CXX_TEMPLATES_INPUTS_CANONICAL_TYPES_H
+
+template<class T>
+struct MagicWrapper {
+  T t;
+  int callGetInt() const {
+    return t.getInt() + 5;
+  }
+};
+
+struct MagicNumber {
+  int getInt() const { return 24; }
```

Comment:
I understand that it is a lot of stuff to rewrite, but I have a hard time keeping track of the various constants in these headers.

WDYT about changing `MagicNumber` to:

```
struct IntWrapper {
  int value;
  int getValue() const { return value; }
};
```

and `callGetInt()` to:

```
int getValuePlusArg(int arg) {
  return t.getValue() + arg;
}
```

This way the correct answer should become obvious in Swift tests.

Reply:
Done, PTAL